From: Andrea Azzarone <azzaronea@gmail.com>
Date: Fri, 16 Nov 2018 18:31:56 +0000
Subject: dash: destroy items's child before tooltip

Destroy the DashItemContainer's child from the same handler as the tooltip. This
will prevent invalid reads when the item is destroyed while its quicklist is
still open.

Bug: https://gitlab.gnome.org/GNOME/gnome-shell/issues/781
Origin: upstream, 3.30.3, commit:ff2fbf5ae40561b403b721e932d1bc5492532156
---
 js/ui/dash.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/js/ui/dash.js b/js/ui/dash.js
index 9b8bfdd..fbc06a7 100644
--- a/js/ui/dash.js
+++ b/js/ui/dash.js
@@ -52,6 +52,8 @@ var DashItemContainer = new Lang.Class({
         this.animatingOut = false;
 
         this.connect('destroy', () => {
+            if (this.child != null)
+                this.child.destroy();
             this.label.destroy();
         });
     },
